diff --git a/generate_self_schema.py b/generate_self_schema.py
index aeb411c6..1cea9453 100644
--- a/generate_self_schema.py
+++ b/generate_self_schema.py
@@ -193,12 +193,7 @@ def all_literal_values(type_: type[core_schema.Literal]) -> list[any]:
 
 
 def eval_forward_ref(type_: Any) -> Any:
-    if sys.version_info < (3, 9):
-        return type_._evaluate(core_schema.__dict__, None)
-    elif sys.version_info < (3, 12, 4):
-        return type_._evaluate(core_schema.__dict__, None, recursive_guard=set())
-    else:
-        return type_._evaluate(core_schema.__dict__, None, type_params=set(), recursive_guard=set())
+    return type_._evaluate(core_schema.__dict__, None, recursive_guard=set())
 
 
 def main() -> None:
